src/tpm2-tss-engine-rand.c

```diff
@@ -40,6 +40,47 @@
 #include "tpm2-tss-engine.h"
 #include "tpm2-tss-engine-common.h"
 
+/** rand seed
+ * @retval 1 on success
+ * @retval 0 on failure
+ */
+static int
+rand_seed(const void *seed, int seed_len)
+{
+    ESYS_CONTEXT *esys_ctx = NULL;
+    TSS2_RC r;
+
+    r = esys_ctx_init(&esys_ctx);
+    ERRchktss(rand_seed, r, goto end);
+
+    TPM2B_SENSITIVE_DATA stir;
+    size_t offset = 0;
+    char* cur_data = (char*)seed;
+
+    static const size_t tpm_random_stir_max_size = 128; 
+    while(offset < (size_t)seed_len) {
+        size_t left = seed_len - offset;
+        // in test, tpm stir seed size beyond 128 will failed with 0x000001d5
```

Comment:
This is in the spec part 3 commands for TPM2_StirRandom, so link to it or mention it here over "it fails, not sure why".